README.md
# Welcome to the Hallowed Grounds :crossed_swords:
#
### About it

This repository is a Marketplace Project signed to me by Yoyo Sensei from Japan.

The main goal is creating a card game with marketplace format, making it possible to trade, draw and discard cards, show hand, build decks, score and eventually chat.

The rules for this project are:

- Only HTML, CSS and JavaScript are allowed;
- It is not allowed Frameworks;
- Bootstrap is allowed.

This same project shall be individually made using Vue.js framework in the near future as part of a programming Yoyo Sensei class.
##

#### The first commit

The first commit is an open repository only with this _README_ file.
##

### Game rules
**Hallowed Grounds** is a card game that involves two or three players who are engaged in a battle of land domination. Each player has their own deck, either one previously constructed or made from a limited pool of cards for the event. All the players start the game with the _Castle_ card on the board, which have 30 as their "_life total_" and loses the game when their life total is reduced to zero. A player can also lose if they must draw from an empty deck.

#### Cards in Hallowed Grounds
There are three types of cards:
- _Castle_
- _Ground_
- _Brave_

The _Castle_ type is the life total of a player, the very first card on the board and it has 30 of life in a total, called _Defense_. This card has two attributes, _Defense_ and _Area_.

The _Ground_ type is the land where the braves battle. This card has four attributes, _Defense_, _Area_, _Capacity_ and _Cost_.

The _Brave_ type is the soldier-like, that can attack or protect. This card has three attributes, _Defense_, _Attack_ and _Cost_.

#### Attributes

There are five card attributes:
- _Attack_
- _Defense_
- _Area_
- _Capacity_
- _Cost_

Basicaly, the _Defense_ attribute is the capabillity to defend an attack. The card is removed from the game when the _Attack_ is greater than the _Defense_. Thus, the card that received the "damage" is discarded to the "discard pile". If the card is defending a castle from a direct attack to it, the card is not discarded because the attack was made against the _Castle_.

The _Area_ attribute is how many _Brave_ cards can be stack over it. Only the Castle and Ground have this attribute. On the other hand the _Cost_ is how much from _Area_ the card take from.

For instance, whether a Ground card has Area of 5, the player only can stack a total of 5 on the card, for example two card with cost of 2 and one with cost of 1.

Lastly, the _Capacity_ attribute is how many cards of that type the player can put on the board. Whether there are two cards in game with Capacity of 2, another one only can be placed on the board if one of these cards is discarded.

#### Attacks

Attacks can be executed against any type of card. If an attack is directed against a Ground card and there are Braves on it, the Braves shall defend the Ground with their Defense attribute. If the Ground is hit by an attack greater than the sum of defenses, the Ground is discarded and the Braves stay inactives until another Ground is put in game under them. In this case, the Braves can receive attacks and cannot be moved to Castle neither to another Ground.

#### Turns

The turns are basicaly when the player of the turn can draw the card and execute movements. The movements can be attacks or movement of troops going from a Ground to another one or to the Castle. The player can place in game as many cards as needed in this moment.

#### Decks and number of cards

There are three types of decks:
- <span style="color:purple">_Dread cards_ (purple)</span>
- <span style="color:darkgreen">_Holy cards_ (green)</span>
- <span style="color:darkred">_Bloody cards_ (red)</span>

The players can choose their deck cards freely, respecting the limit of 50 cards. _(The Castle is not counted for the number of deck cards.)_

Mixed decks are for adventurers. When a mixed deck is created, only the cards with their respective card colours can be place on the _Ground_ card, Brave cards in purple with Ground cards in purple and so on.

To start the match, players need to shuffle the decks and draw 7 cards, each Castle is placed on the board and the player who starts the turn is decided by agreement between the players.




### Copyright

This repository is only for front-end and back-end practices, and it is for non-commercial purposes.

CHANGELOG.md
### Marketplace
# Changelog

All notable changes to this project will be documented in this file.

# [0.0.1] - 2020-02-26
### Added
- Price sum added.
- My Cards section added.

### Changed
No changes.

### Removed
No removal.

---

# [0.0.1] - 2020-02-22
### Added
- Shop added.
- Search bar added.
- Card prices added on the store.

### Changed
No changes.

### Removed
No removal.

---

# [0.0.1] - 2020-02-18
### Added
- Looping sounds were added (burning and snowing).
### Changed
No changes.

### Removed
No removal.

---

# [0.0.1] - 2020-02-16
### Added
- Shopping cart added.
- Cards can be removed when it is clicked.

### Changed
No changes.

### Removed
No removal.

---

# [0.0.1] - 2020-02-15
### Added
No additions.

### Changed
= Card screen modified to show or hide the cards when _Store_ button is clicked.

### Removed
No removal.

---

# [0.0.1] - 2020-02-11
### Added
- Initial game page.
- Fire sparks on initial background page.

### Changed
No changes.

### Removed
No removal.

---

## [0.0.1] - 2020-02-10
### Added
- Snowing background added.
- Castle cards added.

### Changed
No changes.

### Removed
No removal.

---

## [0.0.1] - 2020-02-09
### Added
- Attributes (stats) added to the cards.
- Hovers added.

### Changed
- Description card texts changed.

### Removed
No removal.

---

## [0.0.1] - 2020-02-08
### Added
- First sketch  of the cards in Javascript.

### Changed
No changes.

### Removed
No removal.

---

## [0.0.1] - 2020-02-05
### Added
- First sketch of the cards
- List with ground cards and details.

### Changed
No changes.

### Removed
No removal.

---

js/cards.js
// Card details
const cards = [
  // Dread spot cards
  {
    id: 1,
    name: "Dread Castle",
    type: "Castle",
    description: "A dread fortress, built by the maleficent ancient warlocks to defile the approaching weak souls. Once touched by rune forces, the sense of humanity ends, acquiring corrupted power and mystical insights.",
    area: 12,
    capacity: NaN,
    cost: NaN,
    attack: NaN,
    shop: false,
    hide:false,
    defense: 30,
    price: 2.99,
    colorName: "purple",
    color: "purple",
    img: "url('assets/castledread.jpg')",
  },
  {
    id: 11,
    name: "Dread Spot",
    type: "Ground",
    description: "The spooky ancient place of unholy sacrifices. Warlock masters with mana runes fought the lightforged enemy shields, smiting and purging them to the oblivium.",
    area: 4,
    capacity: 3,
    cost: 0,
    attack: NaN,
    shop: false,
    hide:false,
    defense: 6,
    price: 3.99,
    colorName: "purple",
    color: "purple",
    img: "url('assets/dreadspot1.jpg')",
  },
  {
    id: 12,
    name: "Dread Spot",
    type: "Ground",
    description: "The mists of forsaken souls chase away the light, making the terrain even darker and more gloomy, cornering fire mages with their treacherous tongues, corrupting even the most devout of light.",
    area: 6,
    capacity: 2,
    cost: 0,
    attack: NaN,
    shop: false,
    hide:false,
    defense: 4,
    price: 4.99,
    colorName: "purple",
    color: "purple",
    img: "url('assets/dreadspot2.jpg')",
  },
  {
    id: 13,
    name: "Warlock",
    type: "Brave",
    description: "A vicious warrior that invokes fallen spirits and hits the enemy with their imbued runic punches. The melee attacks cause deep agony and suffering, drilling the flesh and empoisoning it.",
    area: NaN,
    capacity: NaN,
    cost: 2,
    attack: 4,
    shop: false,
    hide:false,
    defense: 5,
    price: 2.99,
    colorName: "purple",
    color: "purple",
    img: "url('assets/warlockdread1.jpg')",
  },
  {
    id: 14,
    name: "Chilling Warlock",
    type: "Brave",
    description: "Chilling warlocks summon demonic energy blasts that stun the most powerful warriors. The dread greenish plasma energy is capable to smash armors and shields with only one hitting.",
    area: NaN,
    capacity: NaN,
    cost: 4,
    attack: 5,
    shop: false,
    hide:false,
    defense: 4,
    price: 3.99,
    colorName: "purple",
    color: "purple",
    img: "url('assets/warlockdread2.jpg')",
  },
  {
    id: 15,
    name: "Stone Witcher",
    type: "Brave",
    description: "The shadow arrows hit the enemies turning them to stone, and making their movements difficult. They are considered the most maleficent rangers of Dread Spot.",
    area: NaN,
    capacity: NaN,
    cost: 2,
    attack: 3,
    shop: false,
    hide:false,
    defense: 4,
    price: 3.99,
    colorName: "purple",
    color: "purple",
    img: "url('assets/witcherdread1.jpg')",
  },
  {
    id: 16,
    name: "Witcher",
    type: "Brave",
    description: "In its shade disguise form, the witcher ambushes behind the enemies, taking them with off guard and totally unprepared.",
    area: NaN,
    capacity: NaN,
    cost: 3,
    attack: 3,
    shop: false,
    hide:false,
    defense: 4,
    price: 5.99,
    colorName: "purple",
    color: "purple",
    img: "url('assets/witcherdread2.jpg')",
  },
  // Holy ground cards
  {
    id: 2,
    name: "Holy Fortress",
    type: "Castle",
    description: "The holy elder's power imbues the rocky building and protects everyone that inhabits the Holy Fortress surroundings. This is the dream place for prospering and growing strong.",
    area: 12,
    capacity: NaN,
    cost: NaN,
    attack: NaN,
    shop: false,
    hide:false,
    defense: 30,
    price: 2.99,
    colorName: "green",
    color: "darkgreen",
    img: "url('assets/castleholy.jpg')",
  },
  {
    id: 21,
    name: "Holy Ground",
    type: "Ground",
    description: "The arcane light brights over the paladins whom wield their swords. This is a place of glory and catharsis after the war and it purifies the soldier spirits.",
    area: 5,
    capacity: 3,
    cost: 0,
    attack: NaN,
    shop: false,
    hide:false,
    defense: 5,
    price: 3.99,
    colorName: "green",
    color: "darkgreen",
    img: "url('assets/holyground1.jpg')",
  },
  {
    id: 22,
    name: "Holy Ground",
    type: "Ground",
    description: "The lightforged shields tinkle across the holy pathway between the ancient empowered rune trees. Light warriors roar bravely with their restored souls in face of battle.",
    area: 5,
    capacity: 5,
    cost: 0,
    attack: NaN,
    shop: false,
    hide:false,
    defense: 3,
    price: 4.99,
    colorName: "green",
    color: "darkgreen",
    img: "url('assets/holyground2.jpg')",
  },
  {
    id: 23,
    name: "Warrior",
    type: "Brave",
    description: "The warriors are trained to bravely kill their enemies with maces, swords, axes or anything else they can wield. The warfare is a constant in their lives, making their souls always alert.",
    area: NaN,
    capacity: NaN,
    cost: 2,
    attack: 2,
    shop: false,
    hide:false,
    defense: 4,
    price: 2.99,
    colorName: "green",
    color: "darkgreen",
    img: "url('assets/warriorholy1.jpg')",
  },
  {
    id: 24,
    name: "Holy Warrior",
    type: "Brave",
    description: "The holy warriors have empowered armors of light that overshadow the path where they tread and confunsing the enemies sights.",
    area: NaN,
    capacity: NaN,
    cost: 3,
    attack: 3,
    shop: false,
    hide:false,
    defense: 5,
    price: 3.99,
    colorName: "green",
    color: "darkgreen",
    img: "url('assets/warriorholy2.jpg')",
  },
  {
    id: 25,
    name: "Holy Hunter",
    type: "Brave",
    description: "Holy hunters are the masters of the archery. They chase and shoot the target quickly as a lightning bolt, neutralize the threat.",
    area: NaN,
    capacity: NaN,
    cost: 2,
    attack: 2,
    shop: false,
    hide:false,
    defense: 3,
    price: 3.99,
    colorName: "green",
    color: "darkgreen",
    img: "url('assets/hunterholy1.jpg')",
  },
  {
    id: 26,
    name: "Wild Hunter",
    type: "Brave",
    description: "The wild hunters tame beasts and use them against the enemies, distracting and hitting them with fire arrows that drill everything along the way.",
    area: NaN,
    capacity: NaN,
    cost: 1,
    attack: 1,
    shop: false,
    hide:false,
    defense: 5,
    price: 5.99,
    colorName: "green",
    color: "darkgreen",
    img: "url('assets/hunterholy2.jpg')",
  },
  // Bloody gorge cards
  {
    id: 3,
    name: "Château Rouge",
    type: "Castle",
    description: "Chateau of the thousand sacrifices, the house of immolation. Demons and night creatures have haunted this place, dazing and confusing the lost souls that approach the hefty structure.",
    area: 12,
    capacity: NaN,
    cost: NaN,
    attack: NaN,
    shop: false,
    hide:false,
    defense: 30,
    price: 2.99,
    colorName: "red",
    color: "darkred",
    img: "url('assets/castlebloody.jpg')",
  },
  {
    id: 31,
    name: "Bloody Gorge",
    type: "Ground",
    description: "The ground was not always covered in blood, yet the gore celebrations were practiced by the insane veterans that stayed aimless wandering. Now these grounds desecrate souls and urge them on with vampiric powers.",
    area: 4,
    capacity: 2,
    cost: 0,
    attack: NaN,
    shop: false,
    hide:false,
    defense: 6,
    price: 3.99,
    colorName: "red",
    color: "darkred",
    img: "url('assets/bloodygorge1.jpg')",
  },
  {
    id: 32,
    name: "Bloody Gorge",
    type: "Ground",
    description: "The bloody moon is even more red when reflecting the imbued vampiric mana ground, becoming the power-well for the forgotten underground creatures.",
    area: 5,
    capacity: 4,
    cost: 0,
    attack: NaN,
    shop: false,
    hide:false,
    defense: 3,
    price: 4.99,
    colorName: "red",
    color: "darkred",
    img: "url('assets/bloodygorge2.jpg')",
  },
  {
    id: 33,
    name: "Blood Berserker",
    type: "Brave",
    description: "A giant familiar demon guides and advises the blood berserker against the enemies. The blood berserker overwhelms everything ahead with the long sword, obliterating even the best forged armors.",
    area: NaN,
    capacity: NaN,
    cost: 2,
    attack: 3,
    shop: false,
    hide:false,
    defense: 4,
    price: 2.99,
    colorName: "red",
    color: "darkred",
    img: "url('assets/berserkerbloody1.jpg')",
  },
  {
    id: 34,
    name: "Demon Berserker",
    type: "Brave",
    description: "Demon berserker is the terror of warfares. Long bloody and fiery wings exempt weapons. It's hard hitting this berserker even when injured.",
    area: NaN,
    capacity: NaN,
    cost: 4,
    attack: 2,
    shop: false,
    hide:false,
    defense: 6,
    price: 3.99,
    colorName: "red",
    color: "darkred",
    img: "url('assets/berserkerbloody2.jpg')",
  },
  {
    id: 35,
    name: "Rogue",
    type: "Brave",
    description: "Rogues come out of nowhere, stabbing behind the enemy with poisoned daggers and venomous blades. If the enemy doesn't die instantly he succumbs in agony.",
    area: NaN,
    capacity: NaN,
    cost: 1,
    attack: 3,
    shop: false,
    hide:false,
    defense: 3,
    price: 3.99,
    colorName: "red",
    color: "darkred",
    img: "url('assets/roguebloody1.jpg')",
  },
  {
    id: 36,
    name: "Dark Rogue",
    type: "Brave",
    description: "Dark rogues stealthy walk within the enemies shadows, easily passing the front and cutting throats before their warcries. Assassination is part of their art.",
    area: NaN,
    capacity: NaN,
    cost: 2,
    attack: 4,
    shop: false,
    hide:false,
    defense: 4,
    price: 5.99,
    colorName: "red",
    color: "darkred",
    img: "url('assets/roguebloody2.jpg')",
  },
]